packages/utils/src/libp2p/dialHelper.ts

```diff
@@ -44,121 +45,189 @@ export type DialResponse =
-  let err: any
+  opts: Required<TimeoutOpts>
+): Promise<
+  | DialResponse
+  | {
+      status: InternalDialStatus.CONTINUE
+    }
+> {
   let struct: Awaited<ReturnType<LibP2P['dialProtocol']>> | null
 
-  let addresses = (libp2p.peerStore.get(destination)?.addresses ?? []).map((addr) => addr.multiaddr.toString())
-
-  // Try to use known addresses
-  if (addresses.length > 0) {
-    try {
-      struct = await libp2p.dialProtocol(destination, protocol, { signal: opts.signal })
-    } catch (_err) {
-      err = _err
+  try {
+    struct = await libp2p.dialProtocol(destination, protocol, { signal: opts.signal })
+  } catch (err) {
+    logError(`Error while dialing ${destination.toB58String()} directly.`)
+    if (err?.message) {
+      logError(`Dial error: ${err.message}`)
     }
   }
 
-  if (struct != null) {
+  // Libp2p's return types tend to change every now and then
+  if (struct != null && struct != undefined) {
     return { status: DialStatus.SUCCESS, resp: struct }
   }
 
   if (opts.signal.aborted) {
     return { status: DialStatus.ABORTED }
   }
 
-  if ((err != null || struct == null) && libp2p.peerRouting._routers.length == 0) {
-    logError(`Could not dial ${destination.toB58String()} directly and libp2p was started without a DHT.`)
-    return { status: DialStatus.DIAL_ERROR, error: err?.message, dhtContacted: false }
-  }
+  return { status: InternalDialStatus.CONTINUE }
+}
 
-  verbose(`could not dial directly${err ? ` (${err.message})` : ''}, looking in the DHT`)
+type DHTResponse = Awaited<ReturnType<LibP2P['peerRouting']['findPeer']>>
 
-  // Try to get some fresh addresses from the DHT
+/**
+ * Performs a DHT query and handles possible errors
+ * @param libp2p Libp2p instance
+ * @param destination which peer to look for
+ * @param opts timeout options
+ */
+async function queryDHT(
+  libp2p: ReducedDHT,
+  destination: PeerId,
+  opts: Required<TimeoutOpts>
+): Promise<DialResponse | { status: InternalDialStatus.CONTINUE; dhtResponse: DHTResponse }> {
   let dhtResponse: DHTResponse
   try {
     // Let libp2p populate its internal peerStore with fresh addresses
     dhtResponse = await libp2p.peerRouting.findPeer(destination, { timeout: DEFAULT_DHT_QUERY_TIMEOUT })
-  } catch (_err) {
-    err = _err
+  } catch (err) {
+    logError(`Error while querying the DHT for ${destination.toB58String()}.`)
+    if (err?.message) {
+      logError(`DHT error: ${err.message}`)
+    }
   }
 
-  if (dhtResponse == null || err != null) {
-    const knownAddresses = libp2p.peerStore.get(destination)?.addresses ?? []
-
-    logError(
-      `Querying the DHT for ${green(destination.toB58String())} failed. Known addresses:\n` +
-        `  ${knownAddresses.length > 0 ? renderPeerStoreAddresses(knownAddresses) : 'No addresses known'}.\n` +
-        `${err ? `\n${err.message}` : ''}`
-    )
+  // Libp2p's return types tend to change every now and then
+  if (dhtResponse == null || dhtResponse == undefined) {
```

Comment:
Similarly here, `dhtResponse == null` should be sufficient

Reply:
Changed